registration/include/pcl/registration/icp.h

```diff
@@ -330,8 +330,8 @@ namespace pcl
       using IterativeClosestPoint<PointSource, PointTarget, Scalar>::transformation_estimation_;
       using IterativeClosestPoint<PointSource, PointTarget, Scalar>::correspondence_rejectors_;
 
-      using Ptr = shared_ptr<IterativeClosestPoint<PointSource, PointTarget, Scalar> >;
-      using ConstPtr = shared_ptr<const IterativeClosestPoint<PointSource, PointTarget, Scalar> >;
+      using Ptr = shared_ptr<IterativeClosestPointWithNormals<PointSource, PointTarget, Scalar> >;
```

Comment:
In order to reduce error and set an example for future code, could I ask for a small refactor?

A code in the spirit of the following would be nice:
```cpp
class SomeClass: BaseClass
{
  using Self = SomeClass;
  using Base = BaseClass;
  public:

  using Ptr = shared_ptr<Self>;
  using ConstPtr = shared_ptr<const Self>;

  using Base::something;
};
```

Reply:
Did not see this in any of the ICP classes. So this would be necessary to fix on all of them.

Reply:
Let's start with one class (this PR), and then modify others (followup PRs)

Reply:
Okay I have committed your suggestions in icp.h.
Should I open up a new pull request or can I add this anywhere here?

Reply:
Thanks. Please create new PRs whenever you can follow up 😄

Reply:
PS: You need to push the changes, they aren't visible on GitHub